spec/models/poll/question_spec.rb

```diff
@@ -86,4 +86,199 @@
+        existing = create(:poll_answer, question: question, author: user, answer: "Before")
+
+        answer = question.find_or_initialize_user_answer(user, answer_text: "After")
+        expect(answer).to eq existing
+        expect(answer.author).to eq user
+        expect(answer.answer).to eq "After"
+      end
+    end
+  end
+
+  describe "scopes" do
+    describe ".for_physical_votes" do
+      it "returns unique and multiple, but not open_ended" do
+        question_unique = create(:poll_question_unique)
+        question_multiple = create(:poll_question_multiple)
+        question_open_ended = create(:poll_question_open)
+
+        result = Poll::Question.for_physical_votes
+
+        expect(result).to match_array [question_unique, question_multiple]
+        expect(result).not_to include question_open_ended
+      end
+    end
+  end
+
+  context "open-ended results" do
+    let(:poll) { create(:poll) }
+    let!(:question_open) { create(:poll_question_open, poll: poll) }
+
+    it "includes voters who didn't answer any questions in blank answers count" do
+      create(:poll_voter, poll: poll)
+
+      expect(question_open.open_ended_blank_answers_count).to eq 1
+      expect(question_open.open_ended_valid_answers_count).to eq 0
+    end
+
+    describe "#open_ended_valid_answers_count" do
+      it "returns 0 when there are no answers" do
+        expect(question_open.open_ended_valid_answers_count).to eq 0
+      end
+
+      it "counts answers" do
+        create(:poll_answer, question: question_open, answer: "Hello")
+        create(:poll_answer, question: question_open, answer: "Bye")
+
+        expect(question_open.open_ended_valid_answers_count).to eq 2
+      end
+    end
+
+    describe "#open_ended_blank_answers_count" do
+      let(:another_question) { create(:poll_question, :yes_no, poll: poll) }
+      let(:option_yes) { another_question.question_options.find_by(title: "Yes") }
+      let(:option_no) { another_question.question_options.find_by(title: "No") }
+
+      it "counts valid participants of the poll who did not answer the open-ended question" do
+        voter_1, voter_2, voter_3 = create_list(:poll_voter, 3, poll: poll)
+
+        create(:poll_answer, question: another_question, author: voter_1.user, option: option_yes)
+        create(:poll_answer, question: another_question, author: voter_2.user, option: option_yes)
+        create(:poll_answer, question: another_question, author: voter_3.user, option: option_yes)
+        create(:poll_answer, question: question_open, author: voter_1.user, answer: "Free text")
+
+        expect(question_open.open_ended_valid_answers_count).to eq 1
+        expect(question_open.open_ended_blank_answers_count).to eq 2
+      end
+
+      it "returns 0 when there are no valid participants in the poll" do
+        expect(question_open.open_ended_blank_answers_count).to eq 0
+      end
+
+      it "counts every user one time even if they answered many questions" do
```

Comment:
The title says "if they answered many questions", but in the test the voter only answers one question :thinking:. What are we trying to test in this scenario?